include/nil/crypto3/math/domains/step_radix2_domain.hpp

```diff
@@ -46,30 +47,43 @@ namespace nil {
             class step_radix2_domain : public evaluation_domain<FieldType, ValueType> {
                 typedef typename FieldType::value_type field_value_type;
                 typedef ValueType value_type;
-
+                typedef std::pair<
```

Comment:
Let's not use a pair here, just 2 separate caches named big_fft_cache, small_fft_cache.